2009-04-24  Aurelien Jarno  <aurelien@aurel32.net>

	* sysdeps/armel/fpu_control.h: If soft-float, don't use
	floating-point registers.

diff --git a/ports/sysdeps/arm/eabi/fpu_control.h b/sysdeps/arm/eabi/fpu_control.h
index 55d7764..30f0f68 100644
--- a/ports/sysdeps/arm/eabi/fpu_control.h
+++ b/ports/sysdeps/arm/eabi/fpu_control.h
@@ -20,6 +20,17 @@
 #ifndef _FPU_CONTROL_H
 #define _FPU_CONTROL_H
 
+#if !defined(_LIBC) && defined(__SOFTFP__)
+
+#define _FPU_RESERVED 0xffffffff
+#define _FPU_DEFAULT  0x00000000
+typedef unsigned int fpu_control_t;
+#define _FPU_GETCW(cw) 0
+#define _FPU_SETCW(cw) do { } while (0)
+extern fpu_control_t __fpu_control;
+
+#else
+
 /* masking of interrupts */
 #define _FPU_MASK_IM	0x00000100	/* invalid operation */
 #define _FPU_MASK_ZM	0x00000200	/* divide by zero */
@@ -48,4 +59,6 @@ typedef unsigned int fpu_control_t;
 /* Default control word set at startup.  */
 extern fpu_control_t __fpu_control;
 
+#endif /* __SOFTFP__ */
+
 #endif /* _FPU_CONTROL_H */
